src/main/drivers/bus_i2c_hal.c

```diff
@@ -206,63 +223,52 @@ void i2cInit(I2CDevice device)
     default:
         break;
     }
-        if (device == I2CINVALID)
-            return;
-
-        i2cDevice_t *i2c;
-        i2c = &(i2cHardwareMap[device]);
-
-        //I2C_InitTypeDef i2cInit;
 
-        IO_t scl = IOGetByTag(i2c->scl);
-        IO_t sda = IOGetByTag(i2c->sda);
+    // Enable RCC
+    RCC_ClockCmd(hardware->rcc, ENABLE);
 
-        IOInit(scl, OWNER_I2C_SCL, RESOURCE_INDEX(device));
-        IOInit(sda, OWNER_I2C_SDA, RESOURCE_INDEX(device));
+    i2cUnstick(scl, sda);
 
-        // Enable RCC
-        RCC_ClockCmd(i2c->rcc, ENABLE);
-
-
-        i2cUnstick(scl, sda);
+    // Init pins
+#ifdef STM32F7
+    IOConfigGPIOAF(scl, pDev->pullUp ? IOCFG_I2C_PU : IOCFG_I2C_NOPU, GPIO_AF4_I2C);
```

Comment:
It depends .. 
- bad case is that it will almost work, causing few bit errors, triggered by special bit sequence.
- clock stretching may make it work, slowing the bus down (by random amount).
- bus performance may be temperature-dependent

With this merged, there is chance board without pullups gets produced and users will complain about software quality ...